lib/Onyx.js

```diff
@@ -1,7 +1,8 @@
 import _ from 'underscore';
-import AsyncStorage from '@react-native-async-storage/async-storage';
 import Str from 'expensify-common/lib/str';
 import lodashMerge from 'lodash/merge';
+import Storage from './storage';
```

Comment:
Instead of using AsyncStorage directly, we're exposing a `Storage` provider that works differently depending on the Platform